src/main/java/com/github/steveice10/mc/protocol/data/MagicValues.java

```diff
@@ -358,7 +358,7 @@
         register(PositionElement.PITCH, 3);
         register(PositionElement.YAW, 4);
 
-        register(GlobalEntityType.LIGHTNING_BOLT, 99);
+        register(GlobalEntityType.LIGHTNING_BOLT, 1);
```

Comment:
Maybe a possible workaround would be this

```suggestion
        register(GlobalEntityType.LIGHTNING_BOLT, 1);
        register(GlobalEntityType.LIGHTNING_BOLT, 99);
        
```

But I think nobody is using the SpawnMobPacket to create lighting bolts...

Reply:
Great, clicking "Resolve" doesn't even submit my comment, I'm not typing that again, so here's the short version:

The only valid type id for ServerSpawnGlobalEntityPacket is 1. The vanilla server never spawns anything else and the client ignores everything else.

We could add a MobType for it but, as you said, I don't think anyone uses SpawnMobPacket for lightning bolts and I'm not even sure the vanilla client would handle it well.

Reply:
I agree.


Keep in mind that MobType also contains alls ObjectTypes, see a9d83ee61b1a4ad48066c3ed4ceff0717eda6796 and ceddf672.

Maybe what we really need is a GlobalEntityType?

Reply:
> Keep in mind that MobType also contains alls ObjectTypes, see a9d83ee and ceddf67.

Yes, I've seen several servers spawn certain objects (e.g. armor stand) using ServerSpawnMobPacket (i.e. MobType) whereas vanilla uses ServerSpawnObjectPacket (i.e. ObjectTypes). No idea why MC makes a distinction between those in the first place, I can only guess that it's for legacy reasons.

> Maybe what we really need is a GlobalEntityType?

Not sure what you're saying. That's quite literally what we have.

Reply:
Wow I total missed that. I thought it was about MobType. Sorry for wasting your time :facepalm: